service/src/main/java/bio/terra/pipelines/dependencies/workspacemanager/WorkspaceManagerService.java

```diff
@@ -0,0 +1,126 @@
+    this.listenerResetRetryTemplate = listenerResetRetryTemplate;
+  }
+
+  public Result checkHealth() {
+    // No access token needed since this is an unauthenticated API.
+    try {
+      workspaceManagerClient
+          .getUnauthenticatedApi()
+          .serviceStatus(); // Workspace Manager's serviceStatus() is a void method
+      return new Result(true, "Workspace Manager is ok");
+    } catch (ApiException e) {
+      return new Result(false, e.getMessage());
+    }
+  }
+
+  public ApiSystemStatusSystems checkHealthApiSystemStatus() {
+    Result healthResult = checkHealth();
+    return new ApiSystemStatusSystems()
+        .ok(healthResult.isOk())
+        .addMessagesItem(healthResult.message());
+  }
+
+  public UUID getWorkspaceStorageResourceId(UUID workspaceId, String accessToken) {
+    ResourceList resourceList =
+        executionWithRetryTemplate(
+            listenerResetRetryTemplate,
+            () ->
+                workspaceManagerClient
+                    .getResourceApi(accessToken)
+                    .enumerateResources(
+                        workspaceId,
+                        null,
+                        null,
+                        ResourceType.AZURE_STORAGE_CONTAINER,
+                        StewardshipType.CONTROLLED));
+
+    // there should be only one AZURE_STORAGE_CONTAINER resource per workspace
+    if (resourceList.getResources().size() != 1) {
+      logger.warn(
+          "Workspace {} has {} AZURE_STORAGE_CONTAINER resources, expected 1",
+          workspaceId,
+          resourceList.getResources().size());
+    }
+
+    return resourceList.getResources().get(0).getMetadata().getResourceId();
+  }
+
+  public static final String READ_PERMISSION_STRING = "r";
+
+  public String getSasTokenForFile(
+      UUID workspaceId, String fullFilePath, String sasPermissions, String accessToken) {
+    UUID resourceId = getWorkspaceStorageResourceId(workspaceId, accessToken);
+    Long sasExpirationDuration =
+        workspaceManagerServerConfiguration.sasExpirationDurationHours() * 60 * 60;
+
+    // extract the blob name from the full file path
+    String blobName = getBlobNameFromHttpUrl(fullFilePath, workspaceId);
+
+    CreatedAzureStorageContainerSasToken createdAzureStorageContainerSasToken =
+        executionWithRetryTemplate(
+            listenerResetRetryTemplate,
+            () ->
+                workspaceManagerClient
+                    .getControlledAzureResourceApi(accessToken)
+                    .createAzureStorageContainerSasToken(
+                        workspaceId,
+                        resourceId,
+                        null,
+                        sasExpirationDuration,
+                        sasPermissions,
+                        blobName));
+    return createdAzureStorageContainerSasToken.getUrl();
+  }
+
+  protected String getBlobNameFromHttpUrl(String blobHttpUrl, UUID workspaceId) {
+    // return the remaining string after the workspaceId
```

Comment:
sorry just cuz im annoying, can we add an example here just cuz its kidna deep in the weeds

Reply:
oh yeah great call. i can copy the one from the test